test/integration_test.go

```diff
@@ -1311,6 +1312,15 @@ func TestPreventDestroy(t *testing.T) {
 	}
 }
 
+func TestPreventDestroyOverride(t *testing.T) {
```

Comment:
Thank you for adding this test case!

Given my comment above, it's prob worth adding an analogous test case, but where:

* The root module sets `prevent_destroy = true`
* The child module doesn't set `prevent_destroy` at all